src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -57,6 +57,7 @@ public override ITestEngineRunner MakeTestRunner(TestPackage package)
             // then defer to the AggregatingTestRunner, which will make
             // the decision on a file by file basis so that each project
             // runs with its own settings.
+            // TODO: What about when ProcessModel is Parallel?
```

Comment:
That is true as AggregatingTestRunner currently works. However, the reason we call AggregatingTestRunner is to allow it to call TestRunnerFactory independently for each of the items on the command line. We do that for two reasons:

1.  So that each separate project will be aggregated separately, creating a test suite in the XML of type Project.

2. Because projects may have different settings specified in the project file and they should be used if not overridden on the command line. Of course, any ProcessModel setting will be ignored if it is set on the command line but there may be other settings.

Maybe we need an intermediary class to deal with this and to (re)simplify AggregatingTestRunner. Something like...
    AggregatingTestRunner <== ParallelTestRunner <== ParallelTestProcessRunner